### ISSUE TITLE
Oil of Sharpness doesn't work on weapons with existing Chance to hit bonus.

### ISSUE THREAD

**Issue by GrayFace** [diasurgical/devilutionX#2221]
**Important information**
Operating system Win 7 x64/ DevilutionX version 1.2.1 / official release

**Describe**
Oil of Sharpness doesn't work on weapons with existing bonus to "Chance to hit". Also, it isn't displayed in item properties when it does work.

**To Reproduce**
1. Load attached save game.
2. Open character stats and inventory.
3. Use the Oil of Sharpness from inventory on the weapon I'm holding.

**Expected behavior**
A change to Hit chance.

**Additional context**
[single_4.zip](https://github.com/diasurgical/devilutionX/files/6705176/single_4.zip)

**Comment by julealgon** [comment]
> Also, it isn't displayed in item properties when it does work.

Please create a separate issue for that. It's unrelated.

**Comment by ThomasChr** [comment]
To me this does not seem to be a bug.
The oil is only applied to the item when the bonus to accuracy the item has is below 50%. The mace you have equipped has a Chance to hit: +90%
See the code:
![image](https://user-images.githubusercontent.com/13014163/128552818-fba8a614-1f6b-495e-8716-37d2421d7580.png)

**Comment by GrayFace** [comment]
That check should happen before the Oil is applied then.

**Comment by ThomasChr** [comment]
I‘m not quite sure how that was intended in vanilla. Maybe they wanted it to vanish without doing anything? So it‘s kind of a mystery?

If we don‘t allow the oil on the weapon (sure I can do that) the user maybe is confused why he cannot use it? Maybe we could use the characters speech, I remember some voice saying „I cannot do that here“ or something similar…

I think there is no right or wrong here and someone with more experience with this project should decide it.

**Comment by julealgon** [comment]
@GrayFace 
> That check should happen before the Oil is applied then. That is, it shouldn't disappear as if it did something.

That's true, but it is a separate issue that applies to most consumable items in the game: for example, if you right click a TP scroll, and then do something else besides casting it, the scroll is consumed and you get nothing in return.

It is a more far-reaching issue that would need to take all consumable items into consideration, not only oils.
